src/chrome/utils/printing.ts

```diff
@@ -4,6 +4,18 @@ function _printClassDescription(this: Function): string {
     return isNotEmpty(this.name) ? `class ${this.name}` : Function.toString.call(this);
 }
 
+function _printInstanceDescription(this: object): string {
+    return isNotEmpty(this.constructor.name) ? `${this.constructor.name}` : 'an anonymous object';
+}
+
 export function printClassDescription(functionConstructor: Function) {
     functionConstructor.toString = _printClassDescription;
+
+    /*
+     * If the class has the default toString method with returns [object Object] change it for our custom method
+     * that returns the class name instead
+     */
+    if (functionConstructor.prototype.toString === Object.prototype.toString) {
```

Comment:
How about just having these class extend a superclass, or implement toString to call a function? This seems like an unusual pattern that could confuse somebody.

Reply:
Agree. Especially because it looks like this will overwrite any existing toString method on the class, which would definitely be surprising behavior. A simple function isn't any more code than a decorator in this case, but much easier to read and understand.

Reply:
Changed to a method on the class...